docs/deployments/domains.mdx

```diff
@@ -2,14 +2,68 @@
 title: Domains
 ---
 
-A domain is the address of your deployment.
+A domain is the address used to access your Globe-hosted project. By default, each project is assigned a unique domain (e.g., `your-project.globe.com`). You can also link a custom domain to your project for a branded experience.
 
-Each deployment, production or preview gets assigned a unique domain that directly points to the specific deployment. If the deployment is not yet ready or successful, visiting a domain will redirect you to the dashboard for that deployment.
+## Linking a Custom Domain
 
-## Production domains
+Custom domains allow you to connect your branded URL (e.g., `your-company.com`) to your Globe-hosted project. This makes your project accessible on a domain of your choice.
 
-Each project gets assigned a unique domain when it is created. Whenever you successfully deploy to the production environment, the production domain will be updated to point to the latest deployment.
+### How It Works
 
-### Custom Domains
+1. Register a domain with an external provider (e.g., Namecheap, GoDaddy, Google Domains).
+2. Configure your DNS settings to point to Globe.
+3. Verify ownership in the Globe Dashboard.
+4. Cloudflare handles DNS management and security, including automatic SSL (HTTPS).
 
-Custom domains are not yet supported. We are working on it.
+Currently, each Globe project can have only one custom domain. To add a new domain, you must remove the existing one first. We will be adding support for multiple domains soon.
+
+## Setting Up Your Custom Domain
+
+### Step 1: Configure DNS with Your Registrar
+
+Before adding a custom domain to Globe, update your DNS settings with your domain provider (e.g., GoDaddy, Google Domains, Cloudflare).
+
+- Add a **CNAME record** pointing your domain to the value provided by Globe.
+- Follow your provider’s setup instructions.
+
+Once updated, add your domain in the Globe Dashboard and verify it.
+
+### Step 2: Add Your Domain in Globe
+
+Each Globe project can have only one linked custom domain at a time. If you want to add a new domain, you must first remove the existing one. We are working on adding support for multiple domains!
+
+1. Go to the **Globe Dashboard**.
+2. Select your project and navigate to **Domains**.
+3. Click **Add Custom Domain**.
+4. Enter your domain name (e.g., `example.com`).
```

Comment:
```suggestion
4. Enter your domain name (e.g., `api.example.com`).
```

@kaziwaseef what are the restrictions on TLD? Wonder if we need to state here

Reply:
This would make it focused on just Backend Apis. The current version has a balance between Flutter web, Dart web, and Dart backend. We could perhaps include multiple examples.

Reply:
There's some restriction on top-level domains we need to be aware of, can't remember what exactly but @kaziwaseef will know

Reply:
I see.